src/serialize/value.rs

```diff
@@ -0,0 +1,906 @@
+            NativeType::BigInt => self.serialize_native::<i64>(typ, cell_writer),
+            NativeType::Text => self.serialize_text(typ, cell_writer),
+            NativeType::Timestamp => {
+                let value = self.extract::<i64>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlTimestamp>(), typ)
+                })?;
+
+                let timestamp = CqlTimestamp(value);
+                timestamp.serialize(typ, cell_writer)
+            }
+            NativeType::Inet => {
+                let value = self.getattr("packed").map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<IpAddr>(), typ)
+                })?;
+
+                let value = value.downcast::<PyBytes>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<IpAddr>(), typ)
+                })?;
+
+                let bytes = value.as_bytes();
+
+                let ip = match bytes.len() {
+                    4 => {
+                        let val: [u8; 4] = bytes.try_into().map_err(|_| {
+                            self.mismatched_type_error(std::any::type_name::<IpAddr>(), typ)
+                        })?;
+                        IpAddr::from(val)
+                    }
+                    16 => {
+                        let val: [u8; 16] = bytes.try_into().map_err(|_| {
+                            self.mismatched_type_error(std::any::type_name::<IpAddr>(), typ)
+                        })?;
+                        IpAddr::from(val)
+                    }
+                    _ => {
+                        return Err(
+                            self.mismatched_type_error(std::any::type_name::<IpAddr>(), typ)
+                        );
+                    }
+                };
+
+                ip.serialize(typ, cell_writer)
+            }
+            NativeType::SmallInt => self.serialize_native::<i16>(typ, cell_writer),
+            NativeType::TinyInt => self.serialize_native::<i8>(typ, cell_writer),
+            NativeType::Time => {
+                let value = self.extract::<i64>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlTime>(), typ)
+                })?;
+
+                let time = CqlTime(value);
+                time.serialize(typ, cell_writer)
+            }
+            NativeType::Timeuuid => {
+                let value = self.getattr("bytes").map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlTimeuuid>(), typ)
+                })?;
+
+                let value = value.downcast::<PyBytes>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlTimeuuid>(), typ)
+                })?;
+
+                let bytes = value.extract::<[u8; 16]>().map_err(|_| {
+                    self.mismatched_type_error(std::any::type_name::<CqlTimeuuid>(), typ)
+                })?;
```

Comment:
💭 We had `packed` attribute in Inet and here we have `bytes`. Where do you take them from?